src/data-workspace/data-entry-cell/entry-field-input.js

```diff
@@ -91,8 +90,21 @@ export function EntryFieldInput({
 
     // todo: inner wrapper?
     const onFocus = useCallback(() => {
-        setHighlightedFieldId({ de, coc })
-    }, [de, coc, setHighlightedFieldId])
+        setHighlightedFieldId({
+            dataElementId: de.id,
+            categoryOptionComboId: coc.id,
+        })
+    }, [de.id, coc.id, setHighlightedFieldId])
+
+    useEffect(() => {
+        return () => {
+            if (highlighted) {
+                setHighlightedFieldId(null)
+            }
+        }
+        // only used to cleanup on unmount - cannot include deps here
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
```

Comment:
Want some input on this.
I've noticed `setHighlightedField(null)` was called in different components around the app. But isn't it easier to just set this null when the component unmounts? I think we would want to reset this whenever the field that is highlighted disappears from the screen, right?

Reply:
As far as I can tell, there are 3 places where we set the highlighted field to null. In 1 case we are setting it to null when the dataset changes, and this could be replaced by setting it to null when the components unmounts. However, in the other two cases, we are setting the highlighted field to null because a filter input is getting focus (see discussion [here](https://dhis2.slack.com/archives/C02NHB92S2H/p1662636916927349?thread_ts=1662630289.253189&cid=C02NHB92S2H)). In these cases the entry field won't unmount.

To conclude: I actually think we can keep this part as it is. It makes sense to have it there.

Reply:
Yeah, it depends on what kind of functionality we want. I could see that it would be beneficial to keep the `highlighted`-field when filtering fields, and if you filter that field away - it would loose focus this way.

I also think it's "safer", in the sense that we might implement something later, or haven't thought of a navigation that should reset the highlighted-field, but doesn't now.

Reply:
@HendrikThePendric I've read through the thread you've linked. Right now I don't understand how you came to the following conclusion:

> Unsetting the active cell when a filter input gets focus should be fairly trivial.

I can't find anything related to this in the messages above. To me, it's quite weird that we hide the sidebar / unset the highlighted item when we focus the highlighted field. What if I just opened the sidebar and then want to filter the results in such a way that the previously highlighted field is still highlighted? Generally it's feels odd that something is happening in the UI just by focussing a search input (in contrast to "changing the filter" value; if we unset the highlighted field when changing the value from an empty string to a non-empty string, that'd be something else)

Reply:
> To me, it's quite weird that we hide the sidebar / unset the highlighted item when we focus the highlighted field.

I guess you mean the `filter field`-here?

I kind of agree, but I do think it's "safer" to clear this when the field unmounts. In the current implementation, we are using `display: none` - so the use-case you highlight (no pun intended) would actually work. I'm not sure if it's actually something we want though - since it could be kind of confusing to have a highlighted-field that is not visible in the form?

Reply:
From slack:

> **Hendrik de Graaf**
> Does this help: https://dhis2.slack.com/archives/C02NHB92S2H/p1662636916927349?thread_ts=1662630289.253189&cid=C02NHB92S2H
>
> > Joe Cooper
> > I think this is almost there. I agree with the premise, and think it's a nice solution. Just testing it though, it's a bit odd when I'm tabbing through a form with sections. The first input in a section is a filter input, so the last input in the previous section gets the grey800 border while the filter input in the next section is active. 

I see, it's about tabbing through the form. IMO we shouldn't unset the active field when focusing the input field because of this:

> To me, it's quite weird that we hide the sidebar / unset the highlighted item when we focus the highlighted field. What if I just opened the sidebar and then want to filter the results in such a way that the previously highlighted field is still highlighted?

It would get tricky to differentiate different reasons why the filter input got focused. And for that reason my preference would be to simply live with the fact that two fields can be highlighted at the same time. If @Birkbjo's current approach keeps the sidebar open & the highlighted item in the state, then I'd prefer to resolve this discussion

cc @cooper-joe I think we need input from you, too

Reply:
If unsetting the highlighted field when focussing on a filter input makes the whole sidebar showing/hiding unnecessarily complicated, then I would also be in favour of just skipping that feature. It was more of a "nice to have" feature to begin with.

@cooper-joe I was thinking we could lighten the outline color of the highlighted-cell-without-focus a bit to make it a bit easier for the user to see that he/she currently has focus on the filter-input, and not the highlighted cell?

Reply:
Yeah, I see the dilemma here. Each option has a downside, so we need to choose which is less-bad. Each option presented as user feedback:
1.  "Why does the last cell in a section have a black border when I tab to the next section? It's confusing."
2.  "Why does the details sidebar close when I tab to the next section? It slows down my workflow." (This PR)

It would be good to implement this in a way that we avoid both of those downsides, but I understand that introduces complexity. So, for now, I'd say living with downside no.1 is less-bad than downside no.2, because downside no.2 could really slow down workflows. So, I suggest dropping the "unset highlighted field when focussing filter" functionality, for now.

There are also other cases that @Birkbjo highlights above, like having a highlighted/focussed field that gets filtered out of the table, so we should ideally land on a solution that solves for all those difficulties too.